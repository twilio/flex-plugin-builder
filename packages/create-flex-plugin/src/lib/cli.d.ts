import { Options } from 'yargs';
export interface CLIArguments {
    accountSid?: string;
    a?: string;
    runtimeUrl?: string;
    r?: string;
    install?: boolean;
    i?: boolean;
    name?: string;
    yarn?: boolean;
    y?: boolean;
    template?: string;
    t?: string;
    typescript?: boolean;
    s?: boolean;
}
export default class CLI {
    static description: string;
    static flags: {
        [key: string]: Options;
    };
    private readonly parser;
    constructor(cwd?: string);
    parse: (...args: string[]) => Promise<void>;
    private init;
}
