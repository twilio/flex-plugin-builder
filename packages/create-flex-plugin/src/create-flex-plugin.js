import execa from 'execa';
import copy from 'copy-template-dir';
import inquirer from 'inquirer';
import { camelCase, upperFirst } from 'lodash';
import path from 'path';
import { promisify } from 'util';
import ora from 'ora';
import fs from 'fs';
import pkg from '../package.json';
import * as log from './logging';

const templatePath = path.resolve(__dirname, '../templates');

const copyDir = promisify(copy);
const moveFile = promisify(fs.rename);

function getPluginJsonContent(config) {
  const plugins = [
    {
      "name": config.pluginClassName,
      "version": "0.0.0",
      "class": config.pluginClassName,
      "requires": [
        {
          "@twilio/flex-ui": config.flexSdkVersion
        }
      ],
      "src": `http://localhost:8080/${config.pluginFileName}.js`
    }
  ]
  if (config.adminPlugin) {
    plugins.push({
      "name": "Admin Plugin",
      "version": "0.7.2",
      "src": "https:\/\/flexperience.twil.io\/assets\/flex.admin.plugin-0.7.2.js"
    });
  }

  return plugins;
}

function parsePluginName(pluginName, config) {
  config.pluginFileName = pluginName;
  config.pluginClassName =
    upperFirst(camelCase(pluginName)).replace('Plugin', '') + 'Plugin';
  return config;
}

function isValidPluginName(name) {
  return name.startsWith('plugin-');
}

async function promptForAccountSid() {
  const response = await inquirer.prompt([
    {
      type: 'input',
      name: 'accountSid',
      message: 'Twilio Flex Account SID',
      validate: x => x.startsWith('AC'),
    },
  ]);
  return response.accountSid;
}

async function promptForAdminPlugin() {
  const response = await inquirer.prompt([
    {
      type: 'confirm',
      name: 'adminPlugin',
      message: 'Do you want to include the Admin Plugin',
      default: true
    }
  ])
  return response.adminPlugin;
}

async function installDependencies(config) {
  const command = config.yarn ? 'yarn' : 'npm';
  const args = ['install'];
  const options = { cwd: config.targetDirectory, shell: process.env.SHELL };
  const { stdout } = await execa(command, args, options);
  return stdout;
}

export default async function createFlexPlugin(config) {
  const pluginName = config.name || '';

  if (!isValidPluginName(pluginName)) {
    log.error('Invalid plugin name. Names need to start with plugin-');
    return process.exit(1);
  }

  config = parsePluginName(pluginName, config);

  if (!config.accountSid) {
    config.accountSid = await promptForAccountSid();
  }

  if (typeof config.adminPlugin === 'undefined') {
    config.adminPlugin = await promptForAdminPlugin();
  }

  config.runtimeUrl = config.runtimeUrl || 'http://localhost:8080';

  config.targetDirectory = path.join(process.cwd(), config.pluginFileName);
  config.flexSdkVersion = pkg.devDependencies['@twilio/flex-ui'];
  config.flexPluginVersion = pkg.devDependencies['flex-plugin'].replace('^', '');
  config.pluginJsonContent = JSON.stringify(getPluginJsonContent(config), null, 2);

  const templateSpinner = ora('Creating project directory');
  try {
    templateSpinner.start();
    const createdFiles = await copyDir(
      templatePath,
      config.targetDirectory,
      config
    );
    await moveFile(
      path.join(config.targetDirectory, 'src/DemoPlugin.js'),
      path.join(config.targetDirectory, `src/${config.pluginClassName}.js`)
    );
    templateSpinner.succeed();
  } catch (err) {
    templateSpinner.fail(err.message);
  }

  if (config.install) {
    const installSpinner = ora('Installing dependencies');
    try {
      installSpinner.start();
      await installDependencies(config);
      installSpinner.succeed();
    } catch (err) {
      installSpinner.fail(err.message);
    }
  }

  log.finalMessage(config);
}
