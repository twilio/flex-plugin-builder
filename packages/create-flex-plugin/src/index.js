"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateFlexPlugin = exports.default = void 0;
/* eslint-disable import/no-unused-modules */
var updateNotifier_1 = require("flex-dev-utils/dist/updateNotifier");
updateNotifier_1.checkForUpdate();
var cli_1 = require("./lib/cli");
Object.defineProperty(exports, "default", { enumerable: true, get: function () { return __importDefault(cli_1).default; } });
var create_flex_plugin_1 = require("./lib/create-flex-plugin");
Object.defineProperty(exports, "CreateFlexPlugin", { enumerable: true, get: function () { return __importDefault(create_flex_plugin_1).default; } });
//# sourceMappingURL=index.js.map