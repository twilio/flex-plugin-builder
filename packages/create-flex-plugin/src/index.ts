/* eslint-disable import/no-unused-modules */
import { checkForUpdate } from '@twilio/flex-dev-utils/dist/updateNotifier';

checkForUpdate();

export { default } from './lib/cli';
export { default as CreateFlexPlugin } from './lib/create-flex-plugin';
