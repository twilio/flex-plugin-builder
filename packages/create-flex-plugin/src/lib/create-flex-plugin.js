"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createFlexPlugin = exports._scaffold = exports._install = void 0;
var fs_1 = __importDefault(require("fs"));
var path_1 = require("path");
var flex_dev_utils_1 = require("flex-dev-utils");
var fs_2 = require("flex-dev-utils/dist/fs");
var strings_1 = require("flex-dev-utils/dist/strings");
var tmp_1 = require("tmp");
var commands_1 = require("./commands");
var finalMessage_1 = __importDefault(require("../prints/finalMessage"));
var validators_1 = __importDefault(require("../utils/validators"));
var templatesRootDir = path_1.resolve(__dirname, '../../templates');
var templateCorePath = path_1.resolve(templatesRootDir, 'core');
var templateJsPath = path_1.resolve(templatesRootDir, 'js');
var templateTsPath = path_1.resolve(templatesRootDir, 'ts');
/**
 * Runs the NPM Installation
 * @param config {FlexPluginArguments}  the configuration
 * @private
 */
var _install = function (config) { return __awaiter(void 0, void 0, void 0, function () {
    return __generator(this, function (_a) {
        return [2 /*return*/, flex_dev_utils_1.progress('Installing dependencies', function () { return __awaiter(void 0, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, commands_1.installDependencies(config)];
                        case 1:
                            _a.sent();
                            return [2 /*return*/, true];
                    }
                });
            }); })];
    });
}); };
exports._install = _install;
/**
 * Creates all the directories and copies the templates over
 *
 * @param config {FlexPluginArguments}  the configuration
 * @private
 */
var _scaffold = function (config) { return __awaiter(void 0, void 0, void 0, function () {
    var dirObject, promise, cleanUp;
    return __generator(this, function (_a) {
        promise = flex_dev_utils_1.progress('Creating project directory', function () { return __awaiter(void 0, void 0, void 0, function () {
            var srcPath, ext;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: 
                    // This copies the core such as public/
                    return [4 /*yield*/, fs_2.copyTemplateDir(templateCorePath, config.targetDirectory, config)];
                    case 1:
                        // This copies the core such as public/
                        _a.sent();
                        srcPath = templateJsPath;
                        if (config.typescript) {
                            srcPath = templateTsPath;
                        }
                        if (!config.template) return [3 /*break*/, 3];
                        dirObject = tmp_1.dirSync();
                        return [4 /*yield*/, commands_1.downloadFromGitHub(config.template, dirObject.name)];
                    case 2:
                        _a.sent();
                        srcPath = dirObject.name;
                        _a.label = 3;
                    case 3: 
                    // This copies the src/ directory
                    return [4 /*yield*/, fs_2.copyTemplateDir(srcPath, config.targetDirectory, config)];
                    case 4:
                        // This copies the src/ directory
                        _a.sent();
                        // Rename plugins
                        if (!dirObject) {
                            ext = config.typescript ? 'tsx' : 'js';
                            fs_1.default.renameSync(path_1.join(config.targetDirectory, "src/DemoPlugin." + ext), path_1.join(config.targetDirectory, "src/" + config.pluginClassName + "." + ext));
                        }
                        return [2 /*return*/, true];
                }
            });
        }); });
        cleanUp = function () {
            if (dirObject) {
                dirObject.removeCallback();
            }
        };
        promise.then(cleanUp).catch(cleanUp);
        return [2 /*return*/, promise];
    });
}); };
exports._scaffold = _scaffold;
/**
 * Creates a Flex Plugin from the {@link FlexPluginArguments}
 * @param config {FlexPluginArguments} the configuration
 */
var createFlexPlugin = function (config) { return __awaiter(void 0, void 0, void 0, function () {
    var _a;
    return __generator(this, function (_b) {
        switch (_b.label) {
            case 0: return [4 /*yield*/, validators_1.default(config)];
            case 1:
                config = _b.sent();
                config = commands_1.setupConfiguration(config);
                // Check folder does not exist
                if (fs_1.default.existsSync(config.targetDirectory)) {
                    throw new flex_dev_utils_1.FlexPluginError(strings_1.singleLineString("Path " + flex_dev_utils_1.logger.coloredStrings.link(config.targetDirectory) + " already exists;", 'please remove it and try again.'));
                }
                return [4 /*yield*/, exports._scaffold(config)];
            case 2:
                // Setup the directories
                if (!(_b.sent())) {
                    throw new flex_dev_utils_1.FlexPluginError('Failed to scaffold project');
                }
                // Add new plugin to .twilio-cli/flex/plugins.json
                return [4 /*yield*/, fs_2.checkPluginConfigurationExists(config.name, config.targetDirectory)];
            case 3:
                // Add new plugin to .twilio-cli/flex/plugins.json
                _b.sent();
                _a = config.install;
                if (!_a) return [3 /*break*/, 5];
                return [4 /*yield*/, exports._install(config)];
            case 4:
                _a = !(_b.sent());
                _b.label = 5;
            case 5:
                // Install NPM dependencies
                if (_a) {
                    flex_dev_utils_1.logger.error('Failed to install dependencies. Please run `npm install` manually.');
                    config.install = false;
                }
                finalMessage_1.default(config);
                return [2 /*return*/];
        }
    });
}); };
exports.createFlexPlugin = createFlexPlugin;
exports.default = exports.createFlexPlugin;
//# sourceMappingURL=create-flex-plugin.js.map