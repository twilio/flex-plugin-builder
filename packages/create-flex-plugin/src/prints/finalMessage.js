"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var flex_dev_utils_1 = require("flex-dev-utils");
var strings_1 = require("flex-dev-utils/dist/strings");
var headline = flex_dev_utils_1.logger.coloredStrings.headline;
/**
 * Prints the final message after the successful creation of a new project
 * @param config
 */
exports.default = (function (config) {
    var installCommand = config.yarn ? 'yarn install' : 'npm install';
    var setupMessage = strings_1.multilineString("" + headline('Setup:'), "$ cd " + config.name + "/", "$ " + installCommand);
    var devMessage = strings_1.multilineString("" + headline('Development:'), "$ cd " + config.name + "/", "$ twilio flex:plugins:start");
    var buildMessage = strings_1.multilineString("" + headline('Build Command:'), "$ cd " + config.name + "/", "$ twilio flex:plugins:build");
    var deployMessage = strings_1.multilineString("" + headline('Deploy Command:'), "$ cd " + config.name + "/", "$ twilio flex:plugins:deploy");
    var message = strings_1.multilineString("Your Twilio Flex Plugin project has been successfully created!", (config.install ? '' : "\n\n " + setupMessage) + "\n", devMessage + "\n", buildMessage + "\n", deployMessage + "\n", 'For more info check the README.md file or go to:', 'âž¡ https://www.twilio.com/docs/flex');
    var columns = (process.stdout.columns || 100) - 14;
    message = flex_dev_utils_1.logger.wrap(message, columns, { hard: true });
    flex_dev_utils_1.boxen.info(message, false);
});
//# sourceMappingURL=finalMessage.js.map