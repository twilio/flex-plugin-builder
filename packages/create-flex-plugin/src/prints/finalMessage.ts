import { logger, boxen, multilineString } from '@twilio/flex-dev-utils';

import { FlexPluginArguments } from '../lib/create-flex-plugin';

const { headline } = logger.coloredStrings;

/**
 * Prints the final message after the successful creation of a new project
 * @param config
 */
export default (config: FlexPluginArguments): void => {
  const installCommand = config.yarn ? 'yarn install' : 'npm install';
  const setupMessage = multilineString(`${headline('Setup:')}`, `$ cd ${config.name}/`, `$ ${installCommand}`);

  const devMessage = multilineString(
    `${headline('Development:')}`,
    `$ cd ${config.name}/`,
    `$ twilio flex:plugins:start`,
  );

  const buildMessage = multilineString(
    `${headline('Build Command:')}`,
    `$ cd ${config.name}/`,
    `$ twilio flex:plugins:build`,
  );

  const deployMessage = multilineString(
    `${headline('Deploy Command:')}`,
    `$ cd ${config.name}/`,
    `$ twilio flex:plugins:deploy`,
  );

  let message = multilineString(
    `Your Twilio Flex Plugin project has been successfully created!`,
    `${config.install ? '' : `\n\n ${setupMessage}`}\n`,
    `${devMessage}\n`,
    `${buildMessage}\n`,
    `${deployMessage}\n`,
    'For more info check the README.md file or go to:',
    'âž¡ https://www.twilio.com/docs/flex',
  );

  const columns = (process.stdout.columns || 100) - 14;
  message = logger.wrap(message, columns, { hard: true });

  boxen.info(message, false);
};
