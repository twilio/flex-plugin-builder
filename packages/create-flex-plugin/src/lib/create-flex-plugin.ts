import fs from 'fs';
import { resolve, join } from 'path';

import { logger, progress, FlexPluginError, singleLineString } from '@twilio/flex-dev-utils';
import { copyTemplateDir, checkPluginConfigurationExists } from '@twilio/flex-dev-utils/dist/fs';
import { dirSync as tmpDirSync, DirResult as TmpDirResult } from 'tmp';

import { setupConfiguration, installDependencies, downloadFromGitHub } from './commands';
import { CLIArguments } from './cli';
import finalMessage from '../prints/finalMessage';
import validate from '../utils/validators';

const templatesRootDir = resolve(__dirname, '../../templates');
const templateCorePath = resolve(templatesRootDir, 'core');
const templateJsPath = resolve(templatesRootDir, 'js');
const templateTsPath = resolve(templatesRootDir, 'ts');

export interface FlexPluginArguments extends CLIArguments {
  name: string;
  targetDirectory: string;
  flexSdkVersion: string;
  pluginScriptsVersion: string;
  pluginClassName: string;
  pluginNamespace: string;
}

/**
 * Runs the NPM Installation
 * @param config {FlexPluginArguments}  the configuration
 * @private
 */
export const _install = async (config: FlexPluginArguments): Promise<boolean> => {
  return progress('Installing dependencies', async () => {
    await installDependencies(config);

    return true;
  });
};

/**
 * Creates all the directories and copies the templates over
 *
 * @param config {FlexPluginArguments}  the configuration
 * @private
 */
export const _scaffold = async (config: FlexPluginArguments): Promise<boolean> => {
  let dirObject: TmpDirResult;

  const promise = progress('Creating project directory', async () => {
    // This copies the core such as public/
    await copyTemplateDir(templateCorePath, config.targetDirectory, config);

    // Get src directory from template URL if provided
    let srcPath = templateJsPath;
    if (config.typescript) {
      srcPath = templateTsPath;
    }
    if (config.template) {
      dirObject = tmpDirSync();
      await downloadFromGitHub(config.template, dirObject.name);
      srcPath = dirObject.name;
    }

    // This copies the src/ directory
    await copyTemplateDir(srcPath, config.targetDirectory, config);

    // Rename plugins
    if (!dirObject) {
      const ext = config.typescript ? 'tsx' : 'js';

      fs.renameSync(
        join(config.targetDirectory, `src/DemoPlugin.${ext}`),
        join(config.targetDirectory, `src/${config.pluginClassName}.${ext}`),
      );
    }

    return true;
  });

  const cleanUp = () => {
    if (dirObject) {
      dirObject.removeCallback();
    }
  };

  promise.then(cleanUp).catch(cleanUp);

  return promise;
};

/**
 * Creates a Flex Plugin from the {@link FlexPluginArguments}
 * @param config {FlexPluginArguments} the configuration
 */
export const createFlexPlugin = async (config: FlexPluginArguments): Promise<void> => {
  config = await validate(config);
  config = setupConfiguration(config);

  // Check folder does not exist
  if (fs.existsSync(config.targetDirectory)) {
    throw new FlexPluginError(
      singleLineString(
        `Path ${logger.coloredStrings.link(config.targetDirectory)} already exists;`,
        'please remove it and try again.',
      ),
    );
  }

  // Setup the directories
  if (!(await _scaffold(config))) {
    throw new FlexPluginError('Failed to scaffold project');
  }

  // Add new plugin to .twilio-cli/flex/plugins.json
  await checkPluginConfigurationExists(config.name, config.targetDirectory);

  // Install NPM dependencies
  if (config.install && !(await _install(config))) {
    logger.error('Failed to install dependencies. Please run `npm install` manually.');
    config.install = false;
  }

  finalMessage(config);
};

export default createFlexPlugin;
