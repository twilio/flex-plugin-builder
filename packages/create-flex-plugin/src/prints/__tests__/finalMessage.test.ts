import { boxen } from '@twilio/flex-dev-utils';

import finalMessage from '../finalMessage';
import { FlexPluginArguments } from '../../lib/create-flex-plugin';

jest.mock('@twilio/flex-dev-utils/dist/logger/lib/boxen');

describe('finalMessage', () => {
  const config: FlexPluginArguments = {
    name: 'plugin-final-message',
    targetDirectory: '/target/directory',
    flexSdkVersion: '1.2.3-flex-sdk',
    pluginScriptsVersion: '1.2.3-plugin-script',
    pluginClassName: 'PluginFinalMessage',
    pluginNamespace: 'PluginFinalMessage',
  };

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('should render an npm setup message to the console', () => {
    let message = '';
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    (boxen as any).info = jest.fn((msg) => (message = msg));

    finalMessage(config);

    expect(boxen.info).toHaveBeenCalledTimes(1);
    expect(message).toMatchSnapshot();
  });

  it('should render a yarn setup message to the console', () => {
    let message = '';
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    (boxen as any).info = jest.fn((msg) => (message = msg));

    finalMessage({ ...config, yarn: true });

    expect(boxen.info).toHaveBeenCalledTimes(1);
    expect(message).toMatchSnapshot();
  });

  it('should render an instruction message skipping the setup step', () => {
    let message = '';
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    (boxen as any).info = jest.fn((msg) => (message = msg));

    finalMessage({ ...config, install: true });

    expect(boxen.info).toHaveBeenCalledTimes(1);
    expect(message).toMatchSnapshot();
  });
});
