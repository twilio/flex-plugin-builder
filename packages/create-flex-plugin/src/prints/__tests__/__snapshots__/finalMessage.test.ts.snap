// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`finalMessage should render a yarn setup message to the console 1`] = `
"Your Twilio Flex Plugin project has been successfully created!


Setup:
$ cd plugin-final-message/
$ yarn install

Development:
$ cd plugin-final-message/
$ twilio flex:plugins:start

Build Command:
$ cd plugin-final-message/
$ twilio flex:plugins:build

Deploy Command:
$ cd plugin-final-message/
$ twilio flex:plugins:deploy

For more info check the README.md file or go to:
➡ https://www.twilio.com/docs/flex"
`;

exports[`finalMessage should render an instruction message skipping the setup step 1`] = `
"Your Twilio Flex Plugin project has been successfully created!


Development:
$ cd plugin-final-message/
$ twilio flex:plugins:start

Build Command:
$ cd plugin-final-message/
$ twilio flex:plugins:build

Deploy Command:
$ cd plugin-final-message/
$ twilio flex:plugins:deploy

For more info check the README.md file or go to:
➡ https://www.twilio.com/docs/flex"
`;

exports[`finalMessage should render an npm setup message to the console 1`] = `
"Your Twilio Flex Plugin project has been successfully created!


Setup:
$ cd plugin-final-message/
$ npm install

Development:
$ cd plugin-final-message/
$ twilio flex:plugins:start

Build Command:
$ cd plugin-final-message/
$ twilio flex:plugins:build

Deploy Command:
$ cd plugin-final-message/
$ twilio flex:plugins:deploy

For more info check the README.md file or go to:
➡ https://www.twilio.com/docs/flex"
`;
